apps/end-tenancy/behaviours/address-lookup.js

```diff
@@ -0,0 +1,250 @@
+      return res.redirect(`${req.url.replace(/\/$/, '')}/postcode`);
+    }
+    return super.get(req, res, callback);
+  }
+
+  getBackLink(req, res, callback) {
+    const backLink = super.getBackLink(req, res, callback);
+    switch (req.params.action) {
+      case 'manual':
+      case 'lookup':
+        return req.url.replace(`/${req.params.action}`, '/postcode');
+      case 'address':
+        return req.url.replace('/address', '/lookup');
+      case 'lookup':
+        return req.url.replace('/lookup', '/postcode');
+      default:
+        return backLink;
+    }
+  }
+
+  getNextStep(req, res, callback) {
+    const step = super.getNextStep(req, res, callback);
+    if (req.params.action === 'postcode') {
+      const newPath = req.sessionModel.get(`${this.options.addressKey}-addresses`) ? '/lookup' : '/address';
+      return req.url.replace('/postcode', newPath);
+    }
+    return step;
+  }
+
+  getValues(req, res, callback) {
+    if (req.params.action === 'manual') {
+      req.sessionModel.unset([
+        `${this.options.addressKey}-postcode`,
+        `${this.options.addressKey}-postcodeApiMeta`
+      ]);
+    } else if (req.params.action === 'lookup') {
+      const addresses = req.sessionModel.get(`${this.options.addressKey}-addresses`);
+      const formattedlist = _.map(_.map(addresses, 'formatted_address'), address => {
+        address = address.split('\n').join(', ');
+        return {
+          value: address,
+          label: address
+        };
+      });
+
+      const count = `${formattedlist.length} address${formattedlist.length > 1 ? 'es' : ''}`;
+      // eslint-disable-next-line max-len
+      req.form.options.fields[`${this.options.addressKey}-select`].options = [{value: '-1', label: count}].concat(formattedlist);
+    }
+    super.getValues(req, res, callback);
+  }
+
+  locals(req, res, callback) {
+    const isManual = req.params.action === 'manual';
+    const locals = super.locals(req, res, callback);
+    const postcode = req.sessionModel.get(`${this.options.addressKey}-postcode`);
+    const section = this.options.route.replace(/^\//, '');
+    return Object.assign({}, locals, {
+      postcodeLabel: req.translate(`fields.${this.options.addressKey}-postcode.label`),
+      postcode,
+      section,
+      route: this.options.route,
+      postcodeApiMessageKey: isManual ?
+        '' :
+        (req.sessionModel.get(`${this.options.addressKey}-postcodeApiMeta`) || {}).messageKey
+    });
+  }
+
+  // eslint-disable-next-line consistent-return
+  process(req, res, callback) {
+    if (req.params.action !== 'postcode') {
+      return super.process(req, res, callback);
+    }
+    const postcode = req.form.values[`${this.options.addressKey}-postcode`];
+    const previousPostcode = req.sessionModel.get(`${this.options.addressKey}-postcode`);
+    if (!postcode
+      || previousPostcode && previousPostcode === postcode) {
+      return callback();
+    }
+
+    if (_.startsWith(postcode, 'BT')) {
```

Comment:
maybe we should add a comment here saying the postcode tool doesn't support northern ireland?